src/linker/runner.py

```diff
@@ -37,13 +38,23 @@ def main(
         ## See above
         "--envvars",
         "foo",
+        "--use-singularity",
+        "--singularity-args",
+        singularity_args,
     ]
     argv.extend(environment_args)
     logger.info(f"Running Snakemake")
     snake_main(argv)
 
 
-def get_environment_args(config, results_dir) -> List[str]:
+def get_singularity_args(input_data: List[Path], results_dir: Path) -> str:
+    input_file_paths = ",".join(file.as_posix() for file in input_data)
+    singularity_args = "--no-home --containall"
+    singularity_args += f" -B /tmp,{results_dir},{input_file_paths}"
```

Comment:
This took me a second to wrap my head around and might benefit from a comment. If I understand correctly, we bind these three paths to be the same inside the container as they are outside. /tmp is so implementations have writable scratch space (we might want to bind a subdir on the outside to this to ensure it is empty).